src/esp/sim/BatchReplayRenderer.cpp

```diff
@@ -244,5 +244,38 @@ void BatchReplayRenderer::doRender(
   renderer_->draw(framebuffer);
 }
 
+#ifdef ESP_BUILD_WITH_CUDA
+const void* BatchReplayRenderer::getCudaColorBufferDevicePointer() {
+  CORRADE_ASSERT(standalone_,
+                 "ReplayBatchRenderer::colorCudaBufferDevicePointer(): can use "
+                 "this function only "
+                 "with a standalone renderer",
+                 nullptr);
+  return static_cast<gfx_batch::RendererStandalone&>(*renderer_)
+      .colorCudaBufferDevicePointer();
+}
+
+const void* BatchReplayRenderer::getCudaDepthBufferDevicePointer() {
+  CORRADE_ASSERT(standalone_,
+                 "ReplayBatchRenderer::getCudaDepthBufferDevicePointer(): can "
+                 "use this function only "
+                 "with a standalone renderer",
+                 nullptr);
+  return static_cast<gfx_batch::RendererStandalone&>(*renderer_)
+      .depthCudaBufferDevicePointer();
+}
+#else
+const void* BatchReplayRenderer::getCudaColorBufferDevicePointer() {
+  ESP_ERROR() << "Failed to retrieve device pointer because CUDA is not "
```

Comment:
you could maybe `#ifdef` inside the functions to not have to repeat the signatures twice